### DISCUSSION TITLE
converter.jar for specific thread

### DISCUSSION THREAD

**Discussion by Benzion18** [async-profiler/async-profiler#760] [General]
Hi,

1. Whenever I run converter.jar jfr2flame, for which thread it creates the flamegraph html?
2. I'd like to convert JFR recording **for specific thread** to **flamegraph html**. Please advise.

**Comment by apangin** [comment]
1. For all threads combined.
2. Run `jfr2flame` with `--threads` option - it will produce per-thread flames, you can also combine it with `--include`/`--exclude` options to filter only threads you are interested in by a regular expression.

For example, to produce a flame graph for just one thread named "subquery-0", use the following command:
```
java -cp converter.jar jfr2flame --threads --include "\[subquery-0.*" profile.jfr profile.html 
```

**Comment by Benzion18** [comment]
[apangin](https://github.com/apangin), thanks!
